core/include/byteorder.h

```diff
@@ -275,6 +275,56 @@ static inline uint64_t NTOHLL(uint64_t v);
 
 /* **************************** IMPLEMENTATION ***************************** */
 
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+
+/**
+ * @brief   Compile time byte-order swapping for short constants
+ */
+#define CONST_HTONS(x)       ((uint16_t)                         \
+                             (((uint16_t)x >> 8) & 0x00FF)     | \
```

Comment:
here and below: parens around macro parameter to let `>>` execute last in case `x` is a term and not a literal.